src/odemis/driver/lakeshore.py

```diff
@@ -0,0 +1,510 @@
+        self.targetTemperature = FloatContinuous(value=self.GetSetpoint(), unit=u"°C", range=[-273, 50], setter=self._set_targetTemperature)
+        self.heating = IntEnumerated(value=self.GetHeaterRange(), choices={0: "Off", 1: "Low", 2: "Medium", 3: "High"},
+                                     setter=self._set_heating)
+
+
+        self._poll_timer = RepeatingTimer(POLL_INTERVAL, self._poll,
+                                                "Lakeshore Polling update")
+
+        self._poll_timer.start()
+
+        # lock the keypad
+        self.LockKeypad(True)
+
+    def terminate(self):
+        self._poll_timer.cancel()
+        time.sleep(0.1)
+
+        if self._serial.isOpen():
+            self.LockKeypad(False)
+
+        with self._ser_access:
+            self._serial.close()
+
+    @staticmethod
+    def _openSerialPort(port, baudrate=57600):
+        """
+        Opens the given serial port the right way for a Power control device.
+        port (string): the name of the serial port (e.g., /dev/ttyUSB0)
+        baudrate (int)
+        return (serial): the opened serial port
+        """
+        ser = serial.Serial(
+            port=port,
+            baudrate=baudrate,
+            bytesize=serial.SEVENBITS,
+            parity=serial.PARITY_ODD,
+            stopbits=serial.STOPBITS_ONE,
+            timeout=2  # s
+        )
+
+        # Purge
+        ser.flush()
+        ser.flushInput()
+
+        # Try to read until timeout to be extra safe that we properly flushed
+        ser.timeout = 0.01
+        while True:
+            char = ser.read()
+            if char == b'':
+                break
+        ser.timeout = 1
+
+        return ser
+
+    def _findDevice(self, ports):
+        """
+        Look for a compatible device
+        ports (str): pattern for the port name
+        baudrate (0<int)
+        return:
+           (str): the name of the port used
+           Note: will also update ._file and ._serial
+        raises:
+            IOError: if no devices are found
+        """
+        # For debugging purpose
+        if ports == "/dev/fake":
+            self._serial = LakeshoreSimulator(timeout=1)
+            self._file = None
+            return ports
+
+        if os.name == "nt":
+            raise NotImplementedError("Windows not supported")
+        else:
+            names = glob.glob(ports)
+
+        for n in names:
+            try:
+                # Ensure no one will talk to it simultaneously, and we don't talk to devices already in use
+                self._file = open(n)  # Open in RO, just to check for lock
+                try:
+                    fcntl.flock(self._file.fileno(), fcntl.LOCK_EX | fcntl.LOCK_NB)  # Raises IOError if cannot lock
+                except IOError:
+                    logging.info("Port %s is busy, will not use", n)
+                    continue
+
+                self._serial = self._openSerialPort(n)
+                return n
```

Comment:
Please check that the device answers something that looks fine from "*IDN?". If not, skip it, and continue searching. There can be so many devices connected via a serial port that it's needed. For an example see piezomotor.py/_findDevice()